cluster-autoscaler/integration/framework.go

```diff
@@ -206,3 +243,115 @@ func (driver *Driver) scaleWorkload(workloadName string, replicas int32) error {
 	}
 	return nil
 }
+
+func (driver *Driver) getFirstNode() (*v1.Node, error) {
+	nodeList, err := driver.targetCluster.Clientset.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return &v1.Node{}, err
+	}
+	sort.Slice(nodeList.Items, func(i, j int) bool {
+		return nodeList.Items[i].ObjectMeta.CreationTimestamp.Before(&nodeList.Items[j].ObjectMeta.CreationTimestamp)
+	})
+	latestNode := nodeList.Items[0]
+	return &latestNode, nil
+}
+
+func (driver *Driver) getLatestNode() (*v1.Node, error) {
+	nodeList, err := driver.targetCluster.Clientset.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return &v1.Node{}, err
+	}
+	sort.Slice(nodeList.Items, func(i, j int) bool {
+		return nodeList.Items[i].ObjectMeta.CreationTimestamp.Before(&nodeList.Items[j].ObjectMeta.CreationTimestamp)
+	})
+	latestNode := nodeList.Items[len(nodeList.Items)-1]
```

Comment:
Are we always sure that there will be atleast one item in the nodelist. Otherwise, this may result in index out of range error. If we aren't sure then better to add a conditional for the same.

Reply:
For the tests that are written until now, we are sure that there will be a node whenever this function is called, but it might not be so in the future. I have added error handling here, and changes are done in the [PR review Commit.](https://github.com/gardener/autoscaler/pull/117/commits/c41b337bb635f451312821595e0d46a28e69f7c6)